src/odemis/util/transform.py

```diff
@@ -270,42 +273,46 @@ def to_physical_space(ji, shape, pixel_size=None):
 
     """
     ji = numpy.asarray(ji)
-    n, m = shape
 
-    if numpy.any(ji < 0):
-        raise IndexError("Negative indices (wrap-around) are not supported.")
-    if numpy.any(ji >= shape):
-        raise IndexError("Index out of range.")
     if ji.shape[-1] != 2:
         raise ValueError("Indices must be 2-dimensional.")
 
     xy = numpy.empty(ji.shape, dtype=float)
-    xy[..., 0] = ji[..., 1] - 0.5 * (m - 1)  # map column-index `i` to x-axis
-    xy[..., 1] = 0.5 * (n - 1) - ji[..., 0]  # map row-index `j` to y-axis
+    xy[..., 0] = ji[..., 1]   # map column-index `i` to x-axis
+    xy[..., 1] = -ji[..., 0]  # map row-index `j` to y-axis
+
+    if shape:
+        # Move the origin to the center of the image.
+        n, m = shape
+        xy[..., 0] -= 0.5 * (m - 1)
+        xy[..., 1] += 0.5 * (n - 1)
 
     if pixel_size is not None:
         xy *= pixel_size
 
     return xy
 
 
-def to_pixel_index(xy, shape, pixel_size=None):
+def to_pixel_index(xy, shape=None, pixel_size=None):
     """
     Converts a coordinate in physical space into an image pixel index.
 
     Inverse of `to_physical_space`. The function `to_pixel_index` converts a
     coordinate in physical space into an image pixel index. The columns of the
     image are aligned with the x-axis, and the rows with the y-axis. The
     direction of the y-axis is opposite to the row-index. The origin of the
-    image pixel index is placed at the center of the top-left pixel.
+    image pixel index is moved to the center of the top-left pixel when
+    provided with the shape of the image. If the shape of the image is not
+    provided the origin is not altered. This is useful when converting a
+    relative displacement (shift).
 
     Parameters
     ----------
     xy : tuple, list of tuples, ndarray
         Physical coordinates, list of coordinates, or array of coordinates. For
         each coordinate the first entry is the `x`-coordinate and the second
         entry is the `y`-coordinate.
-    shape : tuple of ints
+    shape : tuple of ints (optional)
         Shape of the image. The first entry is the number of rows, the second
         entry is the number of columns in the image.
```

Comment:
I think you want to add here:
>If the shape of the image is not provided the origin is not altered.
It is maybe a bit double because it is also in the part above. But because you did it at the function above and I think it makes it more clear it is nice to also add it here.

Reply:
Thanks for noticing. I forgot to add this sentence.